lib/mysqlDao.js

```diff
@@ -356,6 +357,129 @@ class MysqlDao {
     }
 
     /**
+     * @param {object} whereClause
+     * @param {MysqlDaoGetAllOptions}
+     * @param {ReadableOptions} [streamOptions]
+     * @returns {stream.Readable}
+     */
+    stream(whereClause, options, streamOptions = {}) {
+        options = options || {};
+        return this._stream(whereClause, options, {}, streamOptions);
+    }
+
+    /**
+     * @param {object} whereClause
+     * @param {MysqlDaoGetAllOptions}
+     * @returns {stream.Readable}
+     */
+    streamFromMaster(whereClause, options, streamOptions = {}) {
+        options = options || {};
+        return this._stream(whereClause, options, {forceReadFromMaster: true}, streamOptions);
+    }
+
+    /**
+     * @param {string} sql
+     * @param {ReadableOptions} [streamOptions]
+     * @returns {stream.Readable}
+     */
+    streamFromSql(sql, streamOptions = {}) {
+        return this._streamFromSql(sql, {}, streamOptions);
+    }
+
+
+    /**
+     * @param {string} sql
+     * @param {ReadableOptions} [streamOptions]
+     * @returns {stream.Readable}
+     */
+    streamFromSqlRaw(sql, streamOptions = {}) {
+        return this._streamFromSql(sql, {rawResults: true}, streamOptions);
+    }
+
+    /**
+     * @param {string} sql
+     * @param {ReadableOptions} [streamOptions]
+     * @returns {stream.Readable}
+     */
+    streamFromSqlFromMaster(sql, streamOptions) {
+        return this._streamFromSql(sql, {forceReadFromMaster: true}, streamOptions);
+    }
+
+    /**
+     * @param {string} sql
+     * @param {ReadableOptions} [streamOptions]
+     * @returns {stream.Readable}
+     */
+    streamFromSqlFromMasterRaw(sql, streamOptions) {
```

Comment:
Minor, but adding external api functions for all these different permutations of options may be overkill.